### PR TITLE
CIP-1694 | Adding French translation

### PR DESCRIPTION
**Pull request by Hornan7**
Each line has been meticulously translated into French and we are committed to keeping it updated according to each new commit of the original document.

---

([rendered version in branch](https://github.com/Hornan7/CIPs/blob/master/CIP-1694/README.fr.md))

### CODE REVIEW

**Review by rphair — COMMENTED** [review]
@Hornan7 the document content looks great in readability and formatting: thanks to yourself and whoever you are working with to achieve this, with ongoing thanks for offering to keep it updated.

To my knowledge we've not had a parallel language version of a CIP before, so here are my suggestions which @KtorZ @Ryun1 might confirm or debate:

1 - The CIP header data at the beginning of the document, for all translations other than the primary (English) language, should be human- rather than machine-readable.  We are still catching up on our tooling to build generated web sites ([cips.cardano.org](https://cips.cardano.org/) and [developers.cardano.org/docs/governance/cardano-improvement-proposals](https://developers.cardano.org/docs/governance/cardano-improvement-proposals)) which can only happen after we finish standardising header data in a single language (https://github.com/cardano-foundation/CIPs/issues/389)... perhaps someday we will have language independence in the CIP database but we're far from ready to support that.

Therefore any usage of the machine-readable format in the header for multiple languages is going to be ignored, and it doesn't format properly on GitHub for the human reader: so I would recommend these fields be put in Markdown prose (e.g. lists of authors and discussion links) at the beginning of the document: for your convenience and that of your readers.

2 - Your impressive commitment to "keeping it updated" needs to be documented somehow, both before this PR is merged and into the future:

2a) _at least_: the translator name(s), human readable date, and GitHub hash of the English/primary CIP-1694 document which you have translated in your non-English version, or:

2b) _preferably, in my opinion_: a "version history" table which has this information added as a new line of a table or list each time a new translation is submitted as a PR.

At it stands now, there is no notion of "we" other than you, and no notion of "original document" except presumably the latest commit in CIP-1694 listed here (https://github.com/cardano-foundation/CIPs/commits/master/CIP-1694/README.md)... currently only one commit, but that is about to change radically in light of https://github.com/cardano-foundation/CIPs/pull/380#issuecomment-1614916372.

But your brilliant work is just a couple steps away of achieving multi-lingual auditability for this document and I assure you I/we will do everything we can to assist you in that effort. 🤓

**Comment by Hornan7** [comment]
Thank for those nice comments, Its mainly me and Lafleur who worked on this but he doesn't know much about github.

As for the commit history of this document, I'll have to blame my lack of knowledge. I had to copy a new version of the document (without its commits to then perform the PR).
the commit history is in my repository:
(https://github.com/Hornan7/CIP-1694-FR/commits/main)

Our primary objective, for me and Lafleur, was to reproduce the original document exactly line by line in French in the same form. That being said, our motivation is not reduced for all that and we will do our best to rework it for you in the form that will please everyone. 😃👍

**Comment by rphair** [comment]
> the commit history is in my repository:
> (https://github.com/Hornan7/CIP-1694-FR/commits/main)

good to know the source; although unless other editors disagree with point # 2 in https://github.com/cardano-foundation/CIPs/pull/539#pullrequestreview-1511148368, the way forward for this PR  is to link to the latest commit on the _English_ document, not the French one.

**Comment by KtorZ** [comment]
> @rphair: The CIP header data at the beginning of the document, for all translations other than the primary (English) language, should be human- rather than machine-readable.

I (rather strongly) disagree here. The YAML preamble is a structured format, meant to be machine-readable. The **keys** in the mapping are well-defined and shall remain in English. In the same way that by convention in western countries, any comments in source code is written in plain English. 

> @rphair: Therefore any usage of the machine-readable format in the header for multiple languages is going to be ignored

That is another option that I would be quite in favor of. We could omit the preamble altogether on translation, or perhaps, have an alternative preamble for translated content (which I think addresses your other points @rphair) For example:

```yaml
---
cip: 1694
locale: fr-FR
translators:
  - Hornan <contact@able-pool.io>
  - Lafleur <contact@able-pool.io>
revision: abcdef12
---
```

--- 

As for the PR itself, I'll give a read to the French translation this week and provide some feedback.

**Comment by rphair** [comment]
> have an alternative preamble for translated content

yes @KtorZ I like the way the alternative preamble looks & could work in practice.

Question: could it be useful to the human and/or the machine reader to have an alternative preamble with English keys and French values?  At this time the only list of authors and discussion links is in the preamble, so if we omitted the header we'd be omitting this vital context for French readers... which was why I'd suggested putting these lists in the markup.

The revision history on the French document itself (just one English commit ID for each version) would preclude the need to have a human readable "version history" table.  It would be more concise to have this ID in the header as potentially more useful machine-readable data.

So I believe the question is whether to merge the new translation preamble fields in with a preamble having only the values translated... or to use that new translation preamble instead and maybe include the `Authors:` and `Discussions:` some other way.

**Comment by rphair** [comment]
> Translated CIPs, must not include the original preamble.

Sounds good according to the above general guideline.  But as the devil's advocate, what is the user experience of the French reader in this case who is directed to the GitHub pathname for `README.fr.md` and
- wants to follow the Discussion links?  They click on... what?
- wants to view the Authors?  Do they have a URL to click back & look on the original English/primary document header for this?

It can be argued that GitHub makes these things available by going over to the primary file `README.md` but then it appears we are forcing non-English versions to read the English proposal.  This could make sense for a technical / document management but not from a UX point of view.

**Comment by KtorZ** [comment]
@rphair I invite you to continue the conversation on i18n here: https://github.com/cardano-foundation/CIPs/pull/542. Let's keep this PR about the French translation of CIP-1694 and move the conversation about how to support i18n as part of the process.

(Note: I have acknowledged your last statement, and suggest a change compared to my comment just above).

File: CIP-1694/README.fr.md [hunk 1]
```diff
@@ -0,0 +1,1185 @@
+---
+CIP : 1694
+Titre : Un premier pas vers une gouvernance décentralisée on-chain
+Revision : b3e3508a
+Translators :
+    - Mike Hornan <support@able-pool.io>
+    - Alexandre Lafleur <alexandre.lafleur@stm.info>
+Language : fr
```
Comment (KtorZ) [inline CIP-1694/README.fr.md:8]:
```suggestion
CIP: 1694
Titre: Un premier pas vers une gouvernance décentralisée on-chain
Revision: b3e3508a
Translators:
    - Mike Hornan <support@able-pool.io>
    - Alexandre Lafleur <alexandre.lafleur@stm.info>
Language: fr
```

**Comment by Hornan7** [comment]
Let me know if the changes are suitable. I can modify it if necessary if you want me to add additional information such as a list of commit hash of my old updates or some links to discussions in french on cardano forum or twitter for example.

File: CIP-1694/README.fr.md [hunk 2]
```diff
@@ -0,0 +1,1185 @@
+---
+CIP: 1694
+Titre: Un premier pas vers une gouvernance décentralisée on-chain
```
Comment (rphair) [inline CIP-1694/README.fr.md:3]:
```suggestion
Title: Un premier pas vers une gouvernance décentralisée on-chain
```
Agreement above was that keys would be in English.  Both languages look natural for some of these keys (and `Revision` is the same in both languages) so this was an easy one to miss. 🤓

File: CIP-1694/README.fr.md [hunk 3]
```diff
@@ -0,0 +1,1185 @@
+---
```
Comment (rphair) [inline CIP-1694/README.fr.md:1]:
```suggestion
---
Source: https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md
```
We agreed via https://github.com/cardano-foundation/CIPs/pull/542#pullrequestreview-1513216962 to add the URL source so that readers can follow back to the original document, especially for important fields like `Authors` and `Discussions` (and to know to what file the commit hash in `Revision` is referring to).

**Review by rphair — COMMENTED** [review]
@Hornan7 we (w/ @KtorZ) added a new field as per https://github.com/cardano-foundation/CIPs/pull/542 while the i18n requirements are still being reviewed...

**Review by rphair — COMMENTED** [review]
... p.s. resubmitting this 1 item because change was anchored in the wrong place:

File: CIP-1694/README.fr.md (hunk 2 above)
Comment (Hornan7) [reply CIP-1694/README.fr.md:3]: oupsy

File: CIP-1694/README.fr.md [hunk 4]
```diff
@@ -0,0 +1,1185 @@
...
+CIP: 1694
```
[the remark this answers is not in the document]
Comment (Hornan7) [reply CIP-1694/README.fr.md]: Thank you, I'll keep an eye on #542

**Review by rphair — COMMENTED** [review]
(it allowed me to submit my review without the suggested change)

**Comment by Hornan7** [comment]
@rphair maybe i misunderstood, I tought i had to take off "CIP: 1694" and replace it by source.
image

File: CIP-1694/README.fr.md [hunk 5]
```diff
@@ -0,0 +1,1186 @@
...
+Source: https://github.com/cardano-foundation/CIPs/blob/master/CIP-1694/README.md
+Title: Un premier pas vers une gouvernance décentralisée on-chain
+Revision: b3e3508a
```
Comment (rphair) [inline CIP-1694/README.fr.md:5]:
```suggestion
Revision: b3e3508
```

**Review by rphair — COMMENTED** [review]
@KtorZ can you think of any problem with using an 8-digit commit hash here, rather than the 7-digit hashes displayed on the commit history page? https://github.com/cardano-foundation/CIPs/commits/master/CIP-1694/README.md

The problem I can foresee is that searching the commit history page for the 8-digit commit hash won't return a match; while the 7-digit hash will. 😟

File: CIP-1694/README.fr.md (hunk 5 above)
Comment (rphair) [reply CIP-1694/README.fr.md:5]: We could ask it in #542 😅 and we briefly mentioned & ruled out this idea in yesterday's CIP meeting... let's continue to talk about this prospect over there 🙏

Comment (Ryun1) [reply CIP-1694/README.fr.md:5]: oops, you're correct, wrong place to comment that - I will move over to [#542](https://github.com/cardano-foundation/CIPs/pull/542).

File: CIP-1694/README.fr.md [hunk 6]
```diff
@@ -0,0 +1,1186 @@
...
+
+Rien dans cette proposition n’empêche les SPO de devenir des DReps.
+Pourquoi avons-nous des DReps?
+La réponse est que les SPO sont choisis uniquement pour la production de blocs et que tous les SPO ne voudront pas devenir DReps.
+Les électeurs peuvent choisir de déléguer leur vote aux DReps sans avoir à se demander s’ils sont
+également un bon producteur de blocs, et les SPO peuvent choisir de représenter les détenteurs d’Ada ou non.
+
+### Tableau des exigences de ratification
+
+Les conditions énoncées dans le [tableau des conditions de ratification](#requirements) sont expliquées ici.
```
Comment (rphair) [inline CIP-1694/README.fr.md:879]:
```suggestion
Les conditions énoncées dans le [tableau des conditions de ratification](#exigences) sont expliquées ici.
```

**Review by rphair — COMMENTED** [review]
Current translation header requirement is well satisfied: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0001/README.md#translations

and the translation of all literals back to English (in the document as well as the header) is one of the last of the tricking requirements: also looking good.

I had a feeling the section anchors would also be tricky so I double checked them all (they might need to be triple checked)...

(There is one I pointed out in https://github.com/cardano-foundation/CIPs/pull/547 but @Hornan7  that should not be fixed in this PR since that's a problem with the source document that will be fixed in a future commit... _not_ the one that's being translated here... unless also updating the commit hash _after that PR is merged_.  The footnote links to the Shelley and Alonzo ledger specs are also dead here & in the original document, but I will try to address that problem in another PR or issue.)

**Review by rphair — APPROVED** [review]
@KtorZ @Ryun1 @SebastienGllmt I would hope this can be confirmed & merged as a baseline before we start doing too much work through the other CIP-1694 PRs that have been opened in the last week.

**Comment by Hornan7** [comment]
We also noticed these bad links but did not mention them in comments in the original document. (we should have, sorry). That being said, me and lafleur are closely watching #543 #545 #547 and #548  pull requests in review. We will modify the translation when they are merged.

**Review by SebastienGllmt — APPROVED** [review]
